### PR TITLE
fix(atomic): made category facet values hierarchical

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-1677

Old PR: https://github.com/coveo/ui-kit/pull/2029

Visually, nothing has changed, but the CSS is pretty different.

# Deprecations (extracted from the Confluence post)
## Changed
* Changed atomic-category-facet values to be hierarchical.
  * Improves accessibility.
  * Comparison (using the [emmet](https://docs.emmet.io/abbreviations/syntax/) syntax):
    * Before:
      * `ul[part="parents"]`
        * `li > button[part="parent-button"]{All categories}`
        * `li > button[part="parent-button"]{Computers & Tablets}`
        * `li > button[part="parent-button"]{Computer Accessories & Peripherals}`
        * `li > button[part="parent-button"]{Laptop Accessories}`
        * `li > button[part="active-parent"]{Laptop Bags & Cases}`
      * `ul[part="values"]`
        * `li > button[part="value-link"]{Laptop Backpacks}`
        * `li > button[part="value-link"]{Laptop Sleeves}`
        * `li > button[part="value-link"]{Laptop Briefcases}`
        * `li > button[part="value-link"]{Laptop Hard-Shell Cases}`
        * `li > button[part="value-link"]{Laptop Messenger Bags}`
    * After:
      * `[part="parents"]`
        * `button[part="all-categories-button"]{All categories}`
        * `[part="sub-parents"]`
          * `button[part="parent-button"]{Computers & Tablets}`
          * `[part="sub-parents"]`
            * `button[part="parent-button"]{Computer Accessories & Peripherals}`
            * `[part="sub-parents"]`
              * `button[part="parent-button"]{Laptop Accessories}`
              * `[part="sub-parents"]`
                * `button[part="active-parent"]{Laptop Bags & Cases}`
                * `[part="values"]`
                  * `button[part="value-link"]{Laptop Backpacks}`
                  * `button[part="value-link"]{Laptop Sleeves}`
                  * `button[part="value-link"]{Laptop Briefcases}`
                  * `button[part="value-link"]{Laptop Hard-Shell Cases}`
                  * `button[part="value-link"]{Laptop Messenger Bags}`

### CODE REVIEW

**Review by ThibodeauJF — APPROVED** [review]
Looks good!

File: packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx [hunk 1]
```diff
@@ -74,18 +74,18 @@ import {initializePopover} from '../atomic-popover/popover-type';
  * @part search-result-path - The search result path.
  * @part search-highlight - The highlighted query inside the facet values.
  *
- * @part parents - The parent values container.
- * @part all-categories-button - The "View all" button displayed first along the parents.
- * @part parent-button - The clickable parent button.
- * @part active-parent - The non-clickable active parent.
+ * @part parents - The container surrounding the whole hierarchy of values.
+ * @part sub-parents - The container surrounding a part of the hierarchy of values.
```
Comment (jpmarceau) [inline packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx:78]: 'a part' sounds a big vague. Would the following make sense?
```suggestion
 * @part sub-parents - The second-level container, containing a part of the hierarchy of values.
```

File: packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx [hunk 2]
```diff
@@ -74,18 +74,18 @@ import {initializePopover} from '../atomic-popover/popover-type';
...
- * @part values - The facet values child container.
- * @part value-link - The child facet value.
- * @part value-label - The facet value label.
- * @part value-count - The facet value count.
- * @part leaf-value - A facet value with no child value
- * @part node-value - A facet value with children values
+ * @part value-label - The facet value label within a value button.
+ * @part value-count - The facet value count within a value button.
+ * @part leaf-value - A facet value with no child value.
+ * @part node-value - A facet value with children values.
```
Comment (jpmarceau) [inline packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx:88]:
```suggestion
 * @part node-value - A facet value with child values.
```
(if you allow me to be a grammar nerd here: 'child' is an adjective for 'values' in this case, so it shouldn't be conjugated. It's easy to miss, especially for french speakers, so it must've slipped through previously.)

**Review by jpmarceau — APPROVED** [review]
👍

File: packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx (hunk 1 above)
Comment (btaillon) [reply packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx:78]: agreed, but it's not only second-level, technically there's as many `sub-parents` containers embedded in one another as there as the depth of the selected value.

Comment (jpmarceau) [reply packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx:78]: Ah that's interesting, I see why you wrote it this way then. Do you think the following would help:
```suggestion
 * @part sub-parents - The container surrounding a sub-hierarchy of values.
```

**Pull request merged**